matrix/nth_fibonnacci_using_matrix_exponentiation.py

```diff
@@ -0,0 +1,98 @@
+"""
+Implementation of finding nth fibonacci number using matrix exponentiation.
+Time Complexity is about O(log(n)*8)
+As we know
+    f[n] = f[n-1] + f[n-1]
+Converting to matrix,
+    [f(n),f(n-1)] = [[1,1],[1,0]] * [f(n-1),f(n-2)]
+->  [f(n),f(n-1)] = [[1,1],[1,0]]^2 * [f(n-2),f(n-3)]
+    ...
+    ...
+->  [f(n),f(n-1)] = [[1,1],[1,0]]^(n-1) * [f(1),f(0)]
+
+So we just need the n times multiplication of  the matrix [1,1],[1,0]].
+We can decrease the n times multiplication by following  the divide and conquer approach
```

Comment:
Please remove the extra space.

What is the time complexity of the brute force method?  Which one is faster?  How much faster?

Reply:
removed and added